tests/test_parsl_mpi_hello.py

```diff
@@ -155,7 +99,11 @@ def test_run_mpi_hello(config):
         stdout=os.path.join(shared_dir, "parsl_flux_mpi_hello_run.out"),
         stderr=os.path.join(shared_dir, "parsl_flux_mpi_hello_run.err"),
         env=config["environment"],
-        parsl_resource_specification={"num_tasks": 6, "num_nodes": 3},
+        parsl_resource_specification={
+            "num_nodes": 3,  # Number of nodes required for the application instance
+            "ranks_per_node": 2,  # Number of ranks / application elements to be launched per node
+            "num_ranks": 6,  # Number of ranks in total
```

Comment:
If the ordering doesn't matter here, we can reorder for lexicological ordering.
```suggestion
            "num_nodes": 3,  # Number of nodes required for the application instance
            "num_ranks": 6,  # Number of ranks in total
            "ranks_per_node": 2,  # Number of ranks / application elements to be launched per node
```

Reply:
Fixed.